main.js

```diff
@@ -181,14 +182,22 @@ document.querySelector('[data-view]').addEventListener('click', function (e) {
 })
 
 document.querySelector('[data-view = details').addEventListener('click', function (e) {
-  let $button = e.target.closest('.btn-primary')
+  let $button = e.target.closest('.cartButton')
   if ($button !== null) {
     let $currentItem = app.details.item
     app.cart.push($currentItem)
   }
   renderApp(app)
 })
 
+document.querySelector('[data-view = details').addEventListener('click', function (e) {
+  let $button = e.target.closest('.backButton')
```

Comment:
`Element.closest` is intended for querying an element's ancestors. If you are checking the event target itself, just use `Element.getAttribute`.